up_client_socket/cpp/src/SocketUTransport.cpp

```diff
@@ -0,0 +1,340 @@
+    return status;
+  }
+
+  return status;
+}
+
+UStatus SocketUTransport::registerListener(
+    const UUri &topic, const uprotocol::utransport::UListener &listener) {
+  UStatus status;
+  status.set_code(UCode::INTERNAL);
+
+  if (valid_uri(LongUriSerializer::serialize(topic))) {
+    spdlog::debug("SocketUTransport::registerListener():{}, found valid_uri",
+                  __LINE__);
+    status.set_code(UCode::OK);
+    status.set_message("OK");
+    auto uriHash =
+        std::hash<std::string>{}(LongUriSerializer::serialize(topic));
+    std::vector<const uprotocol::utransport::UListener *> &vec =
+        uriToListener[uriHash];
+    if (!vec.empty())
+      vec.push_back(&listener);
+    else {
+      std::vector<const uprotocol::utransport::UListener *> vec1;
+      vec1.push_back(&listener);
+      uriToListener[uriHash] = vec1;
+    }
+  } else {
+    spdlog::warn(
+        "SocketUTransport::registerListener():{}, Received invalid URI",
+        __LINE__);
+    status.set_message("Received invalid URI");
+  }
+
+  return status;
+}
+
+UStatus SocketUTransport::unregisterListener(
+    const UUri &topic, const uprotocol::utransport::UListener &listener) {
+  UStatus status;
+  status.set_code(UCode::INTERNAL);
+
+  if (valid_uri(LongUriSerializer::serialize(topic))) {
+    auto uriHash =
+        std::hash<std::string>{}(LongUriSerializer::serialize(topic));
+    std::vector<const uprotocol::utransport::UListener *> &vec =
+        uriToListener[uriHash];
+
+    auto it = std::find_if(vec.begin(), vec.end(),
+                           [&](const uprotocol::utransport::UListener *item) {
+                             return item == &listener;
+                           });
+    if (it != vec.end()) {
+      spdlog::debug("SocketUTransport::unregisterListener():{}, found listner "
+                    "and removing the same.",
+                    __LINE__);
+      vec.erase(it);
+      status.set_code(UCode::OK);
+      status.set_message("OK");
+    } else {
+      status.set_code(UCode::NOT_FOUND);
+      status.set_message("Listener not found for the given UUri");
+      spdlog::warn("SocketUTransport::unregisterListener():{}, Listener not "
+                   "found for the given UUri",
+                   __LINE__);
+    }
+  } else {
+    status.set_message("Received invalid URI");
+    spdlog::warn(
+        "SocketUTransport::unregisterListener():{}, Received invalid URI",
+        __LINE__);
+  }
+  return status;
+}
+
+void SocketUTransport::timeout_counter(
+    const UUID &req_id,
+    const std::future<uprotocol::rpc::RpcResponse> &resFuture,
+    std::promise<uprotocol::rpc::RpcResponse> &promise, const int timeout) {
+  try {
+    int timeinsecs = timeout / 1000;
+    // std::cout << "SocketUTransport::timeout_counter(),  going to sleep for "
+    // << timeinsecs << " seconds" << std::endl;
+    std::this_thread::sleep_for(std::chrono::seconds(timeinsecs));
```

Comment:
Always use `std::chrono` objects for time
```suggestion
    std::promise<uprotocol::rpc::RpcResponse> &promise, const std::chrono::milliseconds timeout) {
  try {
    std::this_thread::sleep_for(timeout);
```